### DISCUSSION TITLE
Create many and connect throws Unknown arg error

### DISCUSSION THREAD

**Discussion by kdingu** [prisma/prisma#15632] [Q&A] [answered]
edit: prisma ^4.3.1

Hello!

I have just stared working with prisma, and i see this question asked on other discussions as well but i cannot make this work for me...

Models (Vehicle, Attribute) have m:n implicit relationship.
Trying vehicle.createMany and connect them to some existing Attributes.
Prisma generate seems to be ok but still getting this error: Unknown arg `attributes` in data.119.attributes for type VehicleCreateManyInput.

This is the schema for those 2 models:

```
model Attribute {
	...base

	name String
	group AttributeGroup
	type AttributeFieldType
	sort Int? @default(0)
	options AttributeOptions[]
	categories Category[]
	vehicles Vehicle[]

	@@map("attributes")
}

model Vehicle {
	...base

	title String
	description String? @db.LongText
	price Decimal @db.Decimal(24, 2)
	owner_id String
	category_id String
	manufacturer_id String
	model_id String
	attributes Attribute[]
	attribute_option_values VehicleAttributeValues[]
  
	owner User @relation(fields: [owner_id], references: [id])
	category Category @relation(fields: [category_id], references: [id])
	manufacturer Manufacturer @relation(fields: [manufacturer_id], references: [id])
	model Model @relation(fields: [model_id], references: [id])

	@@map("vehicles")
}
```

This is prisma client createMany (simplified) call:

```
vehicles = [
        {
                title: faker.vehicle.manufacturer(),
                description: faker.random.words(30),
                price: faker.commerce.price(1000, 40000),
                owner_id,
                category_id: category.id,
                manufacturer_id,
                model_id,
                attributes: {
                        connect: [
                                {id: 'some-existing-attribute-id'},
                                ...
                        ]
                }
        },
        ...
]
await prisma.vehicle.createMany({data: vehicles});
```

This is part of the error logged:
```
Unknown arg `attributes` in data.119.attributes for type VehicleCreateManyInput. Available args:

type VehicleCreateManyInput {
        id?: String
        created_at?: DateTime
        updated_at?: DateTime
        title: String
        description?: String | Null
        price: Decimal
        owner_id: String
        category_id: String
        manufacturer_id: String
        model_id: String
}

```

**Comment by austincrim** [comment, marked as answer]
Hey @kdingu 👋,

Nested writes within a `createMany` are not yet supported in Prisma. You can always use [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw) for these kinds of queries. 

Feel free to read through this [existing feature request](https://github.com/prisma/prisma/issues/5455) for other potential workarounds and give it a 👍 if you would like to see it added to Prisma.

**Reply by vladi-stevanovic** [reply]
👋 Hello @kdingu did you perhaps have a chance to check Austin's recommendation? Let us know if you still have any questions, or feel free to "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
